core/monitorserver/server.go

```diff
@@ -91,11 +92,12 @@ func (s *Server) getResponseByRevision(epoch int64) (*pb.GetMonitoringResponse,
 	}
 
 	for _, err := range r.Errors {
-		resp.Errors = append(resp.Errors, err.Error())
-	}
-	if len(r.Errors) > 0 {
-		// data to replay the verification steps:
-		resp.ErrorData = r.Response
+		s, ok := status.FromError(err)
```

Comment:
This feels like it can be restructured to remove some of the indentation.  How about:
```
if s, ok := status.FromError(err); ok {
  resp.Errors = append(resp.Errors, s.Proto())
  continue
}
resp.Errors = append(resp.Errors, status.New(codes.Unknown, err.Error()).Proto())
```

Reply:
I like how it shortens it.  Question though..   typically we want to make the indented path the error path, but this makes it the success path.  Is that worth worrying over?

Reply:
Hmm, that's a good point.  Possibly not worth the change in approach just to save a few lines, since as you point out most Go readers expect the indented bit to be early-out for error handling.

Reply:
```
s, ok := status.FromError(err)
if !ok {
  resp.Errors = append(resp.Errors, status.New(codes.Unknown, err.Error()).Proto())
  continue
}
resp.Errors = append(resp.Errors, s.Proto())
```

This reverses it.